couchbase/src/main/scala/akka/stream/alpakka/couchbase/scaladsl/CouchbaseSession.scala

```diff
@@ -31,20 +31,15 @@ object CouchbaseSession {
    * the session is closed.
    */
   def apply(settings: CouchbaseSessionSettings,
-            bucketName: String)(implicit ec: ExecutionContext): Future[CouchbaseSession] = {
-    //wrap CouchbaseAsyncCluster.create up in the Future because it's blocking
-    val asyncCluster: Future[AsyncCluster] =
-      Future(settings.environment match {
-        case Some(environment) =>
-          CouchbaseAsyncCluster.create(environment, settings.nodes: _*)
-        case None =>
-          CouchbaseAsyncCluster.create(settings.nodes: _*)
-      }).map(_.authenticate(settings.username, settings.password))
+            bucketName: String)(implicit ec: ExecutionContext): Future[CouchbaseSession] =
+    openBucket(createClusterClient(settings), disconnectClusterOnClose = true, bucketName)
 
-    asyncCluster
-      .flatMap(c => RxUtilities.singleObservableToFuture(c.openBucket(bucketName), "").map((c, _)))
-      .map { case (c, bucket) => new CouchbaseSessionImpl(bucket, Some(c)) }
-  }
+  /**
+   * Create a session against the given bucket. The life-cycle of the `client` is the user's responsibility.
```

Comment:
Something like "given bucket using a pre-existing cluster client, allowing for it to be shared among multiple CouchbaseSessions."?